libass/ass_cache.c

```diff
@@ -343,36 +350,63 @@ Cache *ass_cache_create(const CacheDesc *desc)
     cache->desc = desc;
     cache->map = calloc(cache->buckets, sizeof(CacheItem *));
     if (!cache->map) {
-        free(cache);
-        return NULL;
+        goto fail;
     }
 
+#if ENABLE_THREADS
```

Comment:
Not doing this since we check errors on init, so this would require a somewhat more complex dummy.

Reply:
```c
#define pthread_mutex_init(c, a) 0
```
is more complex? I can agree in case of `pthread_cond_wait`, but init/destroy is pretty straightforward.

Reply:
I guess I could have `init` be 0 and `destroy` be `do {} while (0)`, but if I made them both 0 (consistent with the actual functions' return types), we'd get compiler warnings about unused expressions.

Reply:
Empty destroy would be fine. Or we could go a long way:
```c
static inline int zero(void)
{
    return 0;
}

#define pthread_mutex_init(m, a) zero()
#define pthread_mutex_destroy(m) zero()
```